ukis_pysat/raster.py

```diff
@@ -73,6 +73,24 @@ def arr(self):
         else:
             return reshape_as_image(self.__arr)
 
+    @arr.setter
+    def arr(self, arr_altered):
+        """array setter"""
+        if not isinstance(arr_altered, np.ndarray):
+            raise TypeError("altered array must be of type np.ndarray")
+
+        if self.dimorder == "last":
+            arr_altered = reshape_as_raster(arr_altered)
+
+        if not arr_altered.shape[-2:] == self.__arr.shape[-2:]:

```

Comment:
If we replace the array I would assume they have the same shape in all dimensions.

Reply:
Not neccessarily the channels dimension. Common array alterations could for example be to extract a band subset from an image or to compute a band index from a multiband image. In both cases the original array would not be altered in its xy dimensions but in the y dimension (channels). I suggest to specifically check for xy and not for z dimension, since xy dimensions are directly related to the geotransformation parameters (z is not).